engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationFieldsBuilder.kt

```diff
@@ -22,18 +23,25 @@ internal object NadelHydrationFieldsBuilder {
         aliasHelper: NadelAliasHelper,
         fieldToHydrate: ExecutableNormalizedField,
         parentNode: JsonNode,
-    ): List<ExecutableNormalizedField> {
+    ): List<CountedBox<ExecutableNormalizedField>> {
         return NadelHydrationInputBuilder.getInputValues(
             instruction = instruction,
             aliasHelper = aliasHelper,
             fieldToHydrate = fieldToHydrate,
             parentNode = parentNode,
-        ).map { args ->
-            makeActorQueries(
+        ).map { boxedArgs ->
+            var counter = 0
+            val args = boxedArgs.mapValues {
+                val hydrationBox = it.value
+                counter += hydrationBox.count
+                hydrationBox.boxedObject
+            }
+            val executableField = makeActorQueries(
```

Comment:
a single executableField can be N hydration ids eg `usersByIds(ids : ["1","2","n"])`

So we have 1 field with a count of N associated with it